manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,357 @@
+            @Override
+            public String schema() {
+               return "";
+            }
+
+            @Override
+            public String table() {
+               return schemaTable[0];
+            }
+         };
+         case 2 -> new ConnectorImplementation(schemaTable);
+         default ->
+            throw new IllegalArgumentException("Unsupported schema format (must be table or schema.table)");
+      };
+   }
+
+   @Override
+   public KeyType keyType() {
+      switch (eagerRule.getKey().getFormat()) {
+         case TEXT:
+            return KeyType.TEXT;
+         case JSON:
+            return KeyType.JSON;
+         case UNRECOGNIZED:
+            throw new IllegalArgumentException("Unsupported KeyType");
+      }
+      throw new IllegalArgumentException("Unsupported KeyType");
+   }
+
+   @Override
+   public String plainSeparator() {
+      return eagerRule.getKey().getKeySeparator();
+   }
+
+   @Override
+   public String selectStatement() {
+      var valueSelect = (eagerRule.getValue() != null && eagerRule.getValue().getValueColumnsCount() > 0)
+            ? String.join(",", eagerRule.getValue().getValueColumnsList())
+            : "*";
+      return "select " + valueSelect + " from " + eagerRule.getTableName();
+   }
+
+   @Override
+   public boolean queryEnabled() {
+      return eagerRule.getQuery().getEnabled();
+   }
+}
+
+// Used only by CacheRuleInformer
+class LazyCacheRuleSpecAdapter implements Rule {
+
+   private final class ConnectorImplementation implements Connector {
+      private final String[] schemaTable;
+
+      private ConnectorImplementation(String[] schemaTable) {
+         this.schemaTable = schemaTable;
+      }
+
+      @Override
+      public String schema() {
+         return schemaTable[0];
+      }
+
+      @Override
+      public String table() {
+         return schemaTable[1];
+      }
+   }
+
+   final LazyCacheRuleSpec lazyRule;
+
+   public LazyCacheRuleSpecAdapter(LazyCacheRuleSpec LazyRule) {
+      this.lazyRule = LazyRule;
+   }
+
+   @Override
+   public Connector connector() {
+      // TODO: how to implement this for lazy rules?
+      // We don't have a table field in the k8s spec
+      return null;
+   }
+
+   @Override
+   public KeyType keyType() {
+      switch (lazyRule.getKey().getFormat()) {
+         case TEXT:
+            return KeyType.TEXT;
+         case JSON:
+            return KeyType.JSON;
+         case UNRECOGNIZED:
+            throw new IllegalArgumentException("Unsupported KeyType");
+      }
+      throw new IllegalArgumentException("Unsupported KeyType");
+   }
+
+   @Override
+   public String plainSeparator() {
+      return lazyRule.getKey().getKeySeparator();
+   }
+
+   @Override
+   public String selectStatement() {
+      return lazyRule.getQuery();
+   }
+
+   @Override
+   public boolean queryEnabled() {
+      // TODO: no query enabled field in the spec.
+      // Is false the correct value?
```

Comment:
For now yes. Once we have a more definitive plan for Query beyond QuickJoin, we can consider adding this field to the CR.